s3store/s3store.go

```diff
@@ -536,3 +549,21 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcPartSize(size *int64) (effectivePartSize int64) {
+	switch {
+	case *size > store.MaxPartSize*10000:
+		effectivePartSize = store.MaxPartSize
+	case *size <= store.MinPartSize:
+		effectivePartSize = store.MinPartSize
+	case *size/10000 < store.MinPartSize:
+		effectivePartSize = store.MinPartSize
+	case int64(math.Mod(float64(*size), 10000)) == 0:
+		effectivePartSize = *size / 10000
+	case *size/9999 >= store.MaxPartSize:
+		effectivePartSize = store.MaxPartSize
+	default:
+		effectivePartSize = *size / 9999
+	}
+	return effectivePartSize
```

Comment:
I have improved on the algorithm, simplified it as much as possible and made sure it will still work, even if Amazon changes current limits in S3 API spec (or if other implementations might use different values).

I have adapted existing tests and improved my own testcase. `go fmt ./s3store` has been run on the code. The patch is also based on the current master which already includes koenvo's patch.

Please let me know, if you need anything else done on this. I am eager to get this merged.